lib/preprocess/raw.js

```diff
@@ -0,0 +1,55 @@
+const { nodeListToArray } = require('../utils')
+
+const regexp = /{{docxRaw [^{}]{0,500}}}/
+
+const supportedParentElements = ['w:r', 'w:p', 'w:tc']

```

Comment:
For now, I've limited the list of supported parent elements. This is needed to know how to many levels up to go: 
https://github.com/jsreport/jsreport-docx/pull/23/files#diff-3cd679edc8ffed05b973faf13bd25249c7b9dbc1d2759755160861a8dd86c2a6R51-R53

That may extended if desired.

Reply:
the idea we had in mind was that we accept param to specify an element, like `w:p`, then you iterate the parent nodes until you find that element, so we can avoid having a fixed list of supported nodes. is that possible? or is there some kind of logic that needs to be done based on the node to be replaced?

Reply:
Implementing this generically as you suggest is possible, the list of fallback wrapping cases just needs to be extended with every possible XML structure: https://github.com/jsreport/jsreport-docx/pull/23/files#diff-5bb476c1d99588325f7846be3b6b13bb91396239a2cec779e0cc2be286724aeaR18-R23

I'll look in this.

Reply:
An alternative may be to not implement fallback wrapping at all and make it the responsibility of the `docxRaw` user to ensure always valid XML is passed.

WDYT?

Reply:
> An alternative may be to not implement fallback wrapping at all and make it the responsibility of the `docxRaw` user to ensure always valid XML is passed.

yes, this is the way to go IMO, this helper is for advanced users so they are in charge of making sure the final docx is valid. we just want to provide the general way to accept the raw xml, if the end result is valid or not then it is the user responsibility and we should not do anything special to prevent this, especially if it makes the implementation a bit harder.

Reply:
I believe I've implemented the requested change. Please let me know if that's what you had in mind.